io-manager.js

```diff
@@ -391,100 +393,100 @@ ioManager.keydown = e => {
           game.drop();
         }
       } else {
+
         // if (!game.dragging) {
-          // _setTransformMode('rotate');
+        // _setTransformMode('rotate');
         // }
+
       }
+
       break;
     }
     case 70: { // F
       e.preventDefault();
       e.stopPropagation();
+
       if (game.canPush()) {
         ioManager.keys.forward = true;
       } else {
         /* if (game.canJumpOff()) {
-          game.jumpOff();
+        game.jumpOff();
         } */
         game.toggleFly();
       }
+
       break;
     }
-    case 71: { // G
-      if (cameraManager.pointerLockElement) {
-        /* if (game.canTry()) {
-          game.menuTry();
-        } */
-      } else {
-        // if (!game.dragging) {
-          // _setTransformMode('translate');
-        // }
-      }
+    case 90: { // Z
+      // if (!e.ctrlKey) {
+      //     if (game.canStartBuild()) {
+      //     game.startBuild('wall');
+      //     } else if (game.canBuild()) {
+      //     game.setBuildMode('wall');
+      //     }
+      // }
+
+      gameManager.dispatchEvent(new CustomEvent('ToggleWorldPanel'));

```

Comment:
This should not be an event. It is an instruction to do something, not a notification of something that did happen already (what events are for).

It should be a function instead.